src/routes/Apps/App/TrainDialogs.tsx

```diff
@@ -257,6 +259,33 @@ class TrainDialogs extends React.Component<Props, ComponentState> {
         }
     }
 
+    async componentDidUpdate(prevProps: Props, prevState: ComponentState) {
+        this.handleQueryParameter(prevProps, prevState)
+    }
+
+    async handleQueryParameter(prevProps: Props, prevState: ComponentState): Promise<void> {
+        const searchParamsPrev = new URLSearchParams(prevProps.location.search)
+        const selectedDialogIdPrev = searchParamsPrev.get("id")
```

Comment:
Would suggest to move "id" to a constant some where so it can't get out of sync with distance calls and it's more strongly typed since `URLSearchParams` isn't. I like that it uses the native browser `URLSearchParams` instead of using library though.

Maybe:
```
const queryParams = {
  id: "id"
}
```

Reply:
good suggestion